### PR TITLE
enable-draggable-tables-for-diagram-organization

### LINKED ISSUE #97: [ENH]: Enable Draggable Tables for Diagram Organization
## 📘  Overview

This improvement request aims to enhance the usability of our model diagram by introducing draggable functionality for the tables. Currently, the tables representing the models and their relationships but they're static, which limits the ability to easily reorganize and view the diagram.

## ✅ Acceptance Criteria

- [ ] Draggable Tables: Allow each table in the diagram to be moved freely by the user.
- [ ] Relationship Consistency: Ensure that as tables are moved, the connections and relationships between them update dynamically to maintain visual consistency.

## 💡 Technical Considerations

- **Performance:** Ensure that the dragging operation remains smooth even when there are multiple tables or complex relationships involved.
- **Edge Cases:** Handle edge cases such as overlapping tables or dragging tables outside the viewable area.
- **Testing:** Thoroughly test with different numbers of tables and relationship types to ensure robustness.

## 🌎 References

Image

#### Conquer the Dojo 🥋

----

⚠️ Do not apply until the ODHack begins.

⚠️ Read our guidelines before applying.

### PR DESCRIPTION
**Pull request by ayomideadeniran**
# 📝 Pull Request Title
Pull Request: Enable Draggable Tables for Diagram Organization (#97)
## 🛠️ Issue
- Closes #97

## 📖 Description
- This PR introduces draggable functionality for tables within the model diagram, enhancing usability and user experience. Users can now freely move tables to better understand or customize the layout of model relationships.

##  Changes made
- Features Implemented
Draggable Tables: Each table in the diagram can now be repositioned via drag-and-drop.

Dynamic Relationship Updates: Connections and relationships between tables update in real-time as tables are moved to maintain diagram accuracy.

Technical Notes
Implemented smooth drag behavior using [relevant library or method, if applicable].

Ensured performance remains optimal with multiple tables and complex relationships.

Implemented boundary checks to prevent dragging tables outside the visible canvas area.

Basic overlap handling implemented to reduce clutter when repositioning.

Testing
Manually tested diagrams with a variety of table counts and connection complexities.

Verified connection lines dynamically adjust during and after dragging.

Tested responsiveness across screen sizes and common interaction patterns.

## 🖼️ Media (screenshots/videos)
-
image

### CODE REVIEW

**Comment by ayomideadeniran** [comment]
https://github.com/KaizeNodeLabs/Tatami/pull/103

**Comment by diegoTech14** [comment]
Hello sir, @ayomideadeniran could you please resolve conflicts?

File: apps/frontend/public/models.json [hunk 1]
```diff
@@ -1,3 +1,82 @@
 {
-  "models": []
+  "models": [
+    {
+      "id": "model_1746114567557",
+      "name": "New Model 1",
```
Comment (diegoTech14) [inline apps/frontend/public/models.json:5]: The idea of the issue is to integrate the drag and drop with the models created, not with default models, please, ensure that your solution meets this criterion

File: apps/frontend/app/app/diagram/code-diagram-section.tsx [hunk 2]
```diff
@@ -238,15 +282,11 @@ export function CodeDiagramSection() {
             </div>
           )
         ) : (
-          <div className="bg-neutral grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6 p-10 overflow-auto h-full">
```
Comment (diegoTech14) [inline apps/frontend/app/app/diagram/code-diagram-section.tsx:241]: Please, ensure that the diagram panel maintains the current style 

image
